helm-chart/metabase/templates/deployment.yaml

```
@@ -0,0 +1,160 @@
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /api/health
+              port: {{ .Values.service.internalPort }}
+            initialDelaySeconds: {{ .Values.livenessProbe.initialDelaySeconds }}
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+          readinessProbe:
+            httpGet:
+              path: /
+              port: {{ .Values.service.internalPort }}
+            initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
+            timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
+            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          volumeMounts:
+          - name: {{ include "metabase.fullname" . }}-configs
+            mountPath: /etc/config
+          - name: {{ include "metabase.fullname" . }}-secrets
+            mountPath: /etc/secret
+            readOnly: true
+          env:
+          {{- if .Values.config.jetty.host }}
+          - name: MB_JETTY_HOST
+            valueFrom:
+              configMapKeyRef:
+                name: {{ include "metabase.fullname" . }}-configs
+                key: MB_JETTY_HOST
+          {{- end }}
+          - name: MB_JETTY_PORT
+            valueFrom:
+              configMapKeyRef:
+                name: {{ include "metabase.fullname" . }}-configs
+                key: MB_JETTY_PORT
+          {{- if and 
+            (ne .Values.config.database.type "h2") 
+            (.Values.secrets.database.host) 
+            (.Values.secrets.database.password) 
+            (.Values.secrets.database.username) 
+            (.Values.secrets.database.port) 
+            (.Values.secrets.database.dbname) 
+          }}
+          - name: MB_DB_TYPE
+            valueFrom:
+              configMapKeyRef:
+                name: {{ include "metabase.fullname" . }}-configs
+                key: MB_DB_TYPE
+          - name: MB_DB_HOST
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "metabase.fullname" . }}-secrets
+                key: MB_DB_HOST
+          - name: MB_DB_PASS
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "metabase.fullname" . }}-secrets
+                key: MB_DB_PASS
+          - name: MB_DB_USER
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "metabase.fullname" . }}-secrets
+                key: MB_DB_USER
+          - name: MB_DB_DBNAME
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "metabase.fullname" . }}-secrets
+                key: MB_DB_DBNAME
+          - name: MB_DB_PORT
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "metabase.fullname" . }}-secrets
+                key: MB_DB_PORT
```

```suggestion
          - name: MB_DB_PORT
            valueFrom:
              secretKeyRef:
                name: {{ include "metabase.fullname" . }}-secrets
                key: MB_DB_PORT
          - name: MB_DB_DBNAME
            valueFrom:
              secretKeyRef:
                name: {{ include "metabase.fullname" . }}-secrets
                key: MB_DB_DBNAME
```

These are less error prone if in the same order as in the if clause. 
Not sure if its necessary, but I has getting errors before changing this.